cmd/upgrade.go

```diff
@@ -27,17 +29,35 @@ var upgradeRuntimeVersion string
 var UpgradeCmd = &cobra.Command{
 	Use:   "upgrade",
 	Short: "Upgrades or downgrades a Dapr control plane installation in a cluster. Supported platforms: Kubernetes",
+	PreRun: func(cmd *cobra.Command, args []string) {
+		viper.BindPFlag("image-registry", cmd.Flags().Lookup("image-registry"))
```

Comment:
Why are you using BindPFlag here and not in the normal init command?

Reply:
@mukundansundar  This is for upgrade case, like `dapr upgrade -k --image-registry`. The normal upgrade command will upgrade based on the whether it is GHCR or docker hub but if someone is using private registry then he shud be able to upgrade from there also.

Reply:
@pravinpushkar My question is in the code. Why in one place you are using BindPFlag when in the other you are not ?

Reply:
@mukundansundar I am using BindPFlag even in init. See here - https://github.com/dapr/cli/blob/8f3e1a9cb90902ef8fe5c67d777b63c290f106ea/cmd/init.go#L50

I just used it like this. Do you think we should switch to local variables